Source/Orts.Simulation/Common/Scripting/Common.cs

```diff
@@ -90,6 +91,11 @@ public Timer(AbstractScriptClass asc)
         {
             CurrentValue = asc.GameTime;
         }
+
+        public Timer(TrainCar car)
+        {
+            CurrentValue = () => (float)car.Simulator.GameTime;
+        }
```

Comment:
When does any content script have access to the TrainCar class? It should be for internal use only.

Reply:
It is used for BatterySwitch and MasterKey classes which are not scriptable.
In the script, you don't have access to TrainCar objects anyway.